### PR TITLE
Match GNU semantics for missing EOF

### PR DESCRIPTION
**Pull request by andrewbaptist**
While the rust coreutils semantics were arguably more correct, they were different than the gnu split semantics when handling a file without a trailing EOF. This patch addresses that difference and allows passing one more GNU test suite.

### CODE REVIEW

**Comment by tertsdiepraam** [comment]
The current `main` branch seems to have the behavior that's in the test case. Could you give an example where it doesn't do the correct thing?

File: tests/by-util/test_split.rs [hunk 1]
```diff
@@ -659,6 +659,22 @@ fn test_line_bytes_no_empty_file() {
     assert!(!at.plus("xak").exists());
 }
 
+#[test]
```
Comment (sylvestre) [inline tests/by-util/test_split.rs:662]: I am confused why this isn't testing the "-io" option too ?

**Comment by sylvestre** [comment]
Well done for:
`Warning: Congrats! The gnu test tests/split/line-bytes is no longer failing!`

File: tests/by-util/test_split.rs (hunk 1 above)
Comment (andrewbaptist) [reply tests/by-util/test_split.rs:662]: The `---io` option is unspecified behavior for GNU split. The code will currently accept and ignore any ---io flags. It is not clear how the flag should be handled as the "chunking" of data is very different.

**Comment by andrewbaptist** [comment]
> Edited comment: I was extremely wrong before, so I just deleted it all.

No problem - this was extremely confusing to me why the GNU split does this. As I mentioned in the comment, the previous Rust behavior was superior and more correct than what I had to change it to. I'm guessing this isn't the only case. It seems like we should have some type of compile-time variable which is "MATCH_GNU" that is enabled for matching GNU behavior but in cases where the behavior is "bad" we don't enable it in "release" builds, or maybe do today, but once we get to 100% matching and the usage of these tools pick up we can find and change all the places. I don't expect anyone is going to be affected by this change in behavior.

**Pull request merged**